### ISSUE TITLE
First Element of Matrix cannot be Empty String

### ISSUE THREAD

**Issue by ThijsvanEssen** [ManimCommunity/manim#1654]
## Description of bug / unexpected behavior
Upon constructing a Matrix if the very first element of the matrix (input[0][0]) is an empty string vectorisation fails in the ```matrix_to_mob_matrix``` function giving ```ValueError: setting an array element with a sequence.``` This is the case for both ```MathTex``` and ```Tex``` as ```element_to_mobject```. Any other position with empty strings is fine. Furthermore, both the Tex transformers support empty strings.

## Expected behavior
A matrix with an empty first element.

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
from typing import List

from manim import *


class Test(Scene):

    def construct(self):

        matrix = Matrix([
            ["", 1],
            [1, 1],
        ])

        self.add(matrix)
```

</details>

## Additional media files

<details><summary>Images/GIFs</summary>

</details>

## Logs
<details><summary>Terminal output</summary>

```
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /home/thijs/.local/lib/python3.8/site-packages/manim/cli/render/commands.py:157 in render        │
│                                                                                                  │
│   154 │   │   for SceneClass in scene_classes_from_file(file):                                   │
│   155 │   │   │   try:                                                                           │
│   156 │   │   │   │   scene = SceneClass()                                                       │
│ ❱ 157 │   │   │   │   scene.render()                                                             │
│   158 │   │   │   except Exception:                                                              │
│   159 │   │   │   │   error_console.print_exception()                                            │
│   160 │   │   │   │   sys.exit(1)                                                                │
│                                                                                                  │
│ /home/thijs/.local/lib/python3.8/site-packages/manim/scene/scene.py:201 in render                │
│                                                                                                  │
│    198 │   │   """                                                                               │
│    199 │   │   self.setup()                                                                      │
│    200 │   │   try:                                                                              │
│ ❱  201 │   │   │   self.construct()                                                              │
│    202 │   │   except EndSceneEarlyException:                                                    │
│    203 │   │   │   pass                                                                          │
│    204 │   │   except RerunSceneException as e:                                                  │
│                                                                                                  │
│ /home/thijs/Documents/Presentation/test.py:10 in construct                                       │
│                                                                                                  │
│    7 │                                                                                           │
│    8 │   def construct(self):                                                                    │
│    9 │   │                                                                                       │
│ ❱ 10 │   │   matrix = Matrix([                                                                   │
│   11 │   │   │   ["", 1],                                                                        │
│   12 │   │   │   [1, 1],                                                                         │
│   13 │   │   ])                                                                                  │
│                                                                                                  │
│ /home/thijs/.local/lib/python3.8/site-packages/manim/mobject/matrix.py:130 in __init__           │
│                                                                                                  │
│   127 │   │   │   raise ValueError(                                                              │
│   128 │   │   │   │   f"{self.__str__()} class requires a two-dimensional array!"                │
│   129 │   │   │   )                                                                              │
│ ❱ 130 │   │   mob_matrix = self.matrix_to_mob_matrix(matrix)                                     │
│   131 │   │   self.organize_mob_matrix(mob_matrix)                                               │
│   132 │   │   self.elements = VGroup(*mob_matrix.flatten())                                      │
│   133 │   │   self.add(self.elements)                                                            │
│                                                                                                  │
│ /home/thijs/.local/lib/python3.8/site-packages/manim/mobject/matrix.py:144 in                    │
│ matrix_to_mob_matrix                                                                             │
│                                                                                                  │
│   141 │   │   │   self.add_background_rectangle()                                                │
│   142 │                                                                                          │
│   143 │   def matrix_to_mob_matrix(self, matrix):                                                │
│ ❱ 144 │   │   return np.vectorize(self.element_to_mobject)(                                      │
│   145 │   │   │   matrix, **self.element_to_mobject_config                                       │
│   146 │   │   )                                                                                  │
│   147                                                                                            │
│                                                                                                  │
│ /home/thijs/.local/lib/python3.8/site-packages/numpy/lib/function_base.py:2113 in __call__       │
│                                                                                                  │
│   2110 │   │   │   vargs = [args[_i] for _i in inds]                                             │
│   2111 │   │   │   vargs.extend([kwargs[_n] for _n in names])                                    │
│   2112 │   │                                                                                     │
│ ❱ 2113 │   │   return self._vectorize_call(func=func, args=vargs)                                │
│   2114 │                                                                                         │
│   2115 │   def _get_ufunc_and_otypes(self, func, args):                                          │
│   2116 │   │   """Return (ufunc, otypes)."""                                                     │
│                                                                                                  │
│ /home/thijs/.local/lib/python3.8/site-packages/numpy/lib/function_base.py:2200 in                │
│ _vectorize_call                                                                                  │
│                                                                                                  │
│   2197 │   │   │   outputs = ufunc(*inputs)                                                      │
│   2198 │   │   │                                                                                 │
│   2199 │   │   │   if ufunc.nout == 1:                                                           │
│ ❱ 2200 │   │   │   │   res = array(outputs, copy=False, subok=True, dtype=otypes[0])             │
│   2201 │   │   │   else:                                                                         │
│   2202 │   │   │   │   res = tuple([array(x, copy=False, subok=True, dtype=t)                    │
│   2203 │   │   │   │   │   │   │    for x, t in zip(outputs, otypes)])                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: setting an array element with a sequence.
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g Windows 10 v2004 or macOS 10.15 (Catalina)): Ubuntu 20.04
- RAM: 8GB
- Python version (`python/py/python3 --version`): 3.8.5
- Installed modules (provide output from `pip list`): 
```Package                Version             
---------------------- --------------------
apturl                 0.5.2               
bcrypt                 3.1.7               
blinker                1.4                 
Brlapi                 0.7.0               
certifi                2019.11.28          
chardet                3.0.4               
click                  8.0.1               
click-default-group    1.2.2               
cloup                  0.7.1               
colorama               0.4.3               
colour                 0.1.5               
command-not-found      0.3                 
commonmark             0.9.1               
cryptography           2.8                 
cupshelpers            1.0                 
dbus-python            1.2.16              
decorator              5.0.9               
defer                  1.0.6               
distro                 1.4.0               
distro-info            0.23ubuntu1         
duplicity              0.8.12.0            
entrypoints            0.3                 
fasteners              0.14.1              
future                 0.18.2              
glcontext              2.3.4               
httplib2               0.14.0              
idna                   2.8                 
keyring                18.0.1              
language-selector      0.1                 
launchpadlib           1.10.13             
lazr.restfulclient     0.14.2              
lazr.uri               1.0.3               
lockfile               0.12.2              
louis                  3.12.0              
macaroonbakery         1.3.1               
Mako                   1.1.0               
manim                  0.7.0               
ManimPango             0.3.0               
mapbox-earcut          0.12.10             
MarkupSafe             1.1.0               
moderngl               5.6.4               
moderngl-window        2.3.0               
monotonic              1.5                 
multipledispatch       0.6.0               
netifaces              0.10.4              
networkx               2.5.1               
numpy                  1.20.3              
oauthlib               3.1.0               
olefile                0.46                
paramiko               2.6.0               
pexpect                4.6.0               
Pillow                 7.0.0               
pip                    20.0.2              
protobuf               3.6.1               
pycairo                1.20.1              
pycups                 1.9.73              
pydub                  0.25.1              
pyglet                 1.5.17              
Pygments               2.9.0               
PyGObject              3.36.0              
PyJWT                  1.7.1               
pymacaroons            0.13.0              
PyNaCl                 1.3.0               
pyRFC3339              1.1                 
pyrr                   0.10.3              
python-apt             2.0.0+ubuntu0.20.4.5
python-dateutil        2.7.3               
python-debian          0.1.36ubuntu1       
pytz                   2019.3              
pyxdg                  0.26                
PyYAML                 5.3.1               
reportlab              3.5.34              
requests               2.22.0              
requests-unixsocket    0.2.0               
rich                   10.2.2              
scipy                  1.6.3               
SecretStorage          2.3.1               
setuptools             45.2.0              
simplejson             3.16.0              
six                    1.14.0              
systemd-python         234                 
tqdm                   4.61.0              
typing-extensions      3.10.0.0            
ubuntu-advantage-tools 27.0                
ubuntu-drivers-common  0.0.0               
ufw                    0.36                
unattended-upgrades    0.1                 
urllib3                1.25.8              
usb-creator            0.3.7               
wadllib                1.3.3               
watchdog               2.1.2               
wheel                  0.34.2              
xkit                   0.0.0     
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020): TeX Live 2019/Debian
+ Installed LaTeX packages:
(running on Debian, switching to user mode!)

</details>
<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
ffmpeg version 4.2.4-1ubuntu0.1 Copyright (c) 2000-2020 the FFmpeg developers
built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
configuration: --prefix=/usr --extra-version=1ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
libavutil      56. 31.100 / 56. 31.100
libavcodec     58. 54.100 / 58. 54.100
libavformat    58. 29.100 / 58. 29.100
libavdevice    58.  8.100 / 58.  8.100
libavfilter     7. 57.100 /  7. 57.100
libavresample   4.  0.  0 /  4.  0.  0
libswscale      5.  5.100 /  5.  5.100
libswresample   3.  5.100 /  3.  5.100
libpostproc    55.  5.100 / 55.  5.100
```
</details>

## Additional comments
Pretty sure this would be a bug. Since an empty string is supported every where else. However, was not able to find out without question, if this is intended or undefined behavior I am curious to find out!

**Comment by kilacoda** [comment]
Thanks for reporting this. I looked into it, and found the error causing line:
https://github.com/ManimCommunity/manim/blob/a8d8c18e0148a651121715fa90b75c6a2cdfb9a9/manim/mobject/matrix.py#L143-L146

Seems to be something related to the [`otypes`](https://numpy.org/doc/stable/reference/generated/numpy.vectorize.html#numpy-vectorize) parameter, and is fixed by changing the line in question to:

```py
 def matrix_to_mob_matrix(self, matrix): 
     return np.vectorize(self.element_to_mobject, otypes = [object])( 
         matrix, **self.element_to_mobject_config 
     ) 
```

Do you want to make a PR for this fix? Would make for a good first contribution. It's fine if you don't, I could do it otherwise.

**Comment by kilacoda** [comment]
Oh, btw here's the output. Is this similar to what you were trying to acheive?
![image](https://user-images.githubusercontent.com/65204531/121327423-0e929500-c931-11eb-8b75-d564ef16ac73.png)

**Comment by Patil2099** [comment]
I would love to take this up.

**Comment by ThijsvanEssen** [comment]
Hi,

Precisely what I want. I would love to make a Pull Request for this. I will figure out how to properly contribute via the guide and will investigate what you fix means completely! Thanks so much :)

(I see it might already be fixed, in that case ill investigate what the changes mean exactly ;))

**Comment by Patil2099** [comment]
You should keep this open I would say @ThijsvanEssen till my pr is merged.

**Comment by PhotonSpheres** [comment]
Hey there! Thanks for the issue. I have rewritten parts of the matrix module in #1636 and this issue is also fixed there. As soon as we merge #1636 this will work just fine.

**Comment by jsonvillanueva** [comment]
Fixed in #1636
